### PR TITLE
Fix predefined metrics not match in monitoring

### PR DESCRIPTION
**Pull request by thxCode**
**Issue:**
https://github.com/rancher/rancher/issues/18624

**Related PR:**
https://github.com/rancher/system-charts/pull/26

### CODE REVIEW

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 1]
```diff
@@ -1586,7 +1587,7 @@ metadata:
     component: cluster
     details: "false"
     level: cluster
-    metric: cpu-usage-seconds-sum-rate
+    graph: cpu-usage-seconds-sum-rate
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: why we need to change label? Metric label and component label combine to the name. If a graph have multiple metric, you can add graph label to select them all.

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 2]
```diff
@@ -3441,7 +3442,6 @@ metadata:
     source: rancher-monitoring
 spec:
   expression: sum(rate(node_network_transmit_bytes_total{device!~"lo|veth.*|docker.*|flannel.*|cali.*|cbr.*",instance=~"$instance"}[5m]))
-     * 8 / 1024
   legendFormat: Transmit
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: Could you check with Logan whether the UI needs an update?

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 3]
```diff
@@ -2422,134 +2425,8 @@ spec:
...
-    metric: disk-io-reads-bytes-sum-rate
-    source: rancher-monitoring
-spec:
-  expression: sum(rate(container_fs_reads_bytes_total{namespace=~"$namespace",pod_name=~"$podName",
-    container_name=~"$containerName"}[5m]))
-  legendFormat: Read([[container_name]])
-  description: container disk io reads bytes sum rate
----
-#### Container Metrics ####
-
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: move the deleted container metric code to another commit

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 4]
```diff
@@ -2714,7 +2714,7 @@ metadata:
     metric: disk-wal-fsync-duration-seconds-sum-quantile
     source: rancher-monitoring
 spec:
-  expression: sum(histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m])))
+  expression: histogram_quantile(0.99, sum(rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m])))
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: ```sum(histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m])))``` the sum is for sum value from mutiple nodes inside the cluster. This one ```histogram_quantile(0.99, sum(rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m])))```  throw error no datapoint

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 5]
```diff
@@ -2732,8 +2732,7 @@ metadata:
     metric: disk-wal-fsync-duration-seconds-sum-quantile
     source: rancher-monitoring
 spec:
-  expression: sum(histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m])))
-    by (instance)
+  expression: histogram_quantile(0.99, sum(rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m])) by (instance, le))
   legendFormat: WAL fsync([[instance]])
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: same here.

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 6]
```diff
@@ -2784,7 +2783,7 @@ metadata:
     metric: disk-commit-duration-seconds-sum-quantile
     source: rancher-monitoring
 spec:
-  expression: sum(histogram_quantile(0.99, rate(etcd_disk_backend_commit_duration_seconds_bucket[5m])))
+  expression: histogram_quantile(0.99, sum(rate(etcd_disk_backend_commit_duration_seconds_bucket[5m])))
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: same here.

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 7]
```diff
@@ -2802,8 +2801,7 @@ metadata:
     metric: disk-commit-duration-seconds-sum-quantile
     source: rancher-monitoring
 spec:
-  expression: sum(histogram_quantile(0.99, rate(etcd_disk_backend_commit_duration_seconds_bucket[5m])))
-    by (instance)
+  expression: histogram_quantile(0.99, sum(rate(etcd_disk_backend_commit_duration_seconds_bucket[5m])) by (instance, le))
   legendFormat: DB fsync([[instance]])
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: same here.

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 8]
```diff
@@ -2819,7 +2817,7 @@ metadata:
     metric: grpc-request-slow-quantile
     source: rancher-monitoring
 spec:
-  expression: sum(histogram_quantile(0.99, sum(rate(grpc_server_handling_seconds_bucket{grpc_type="unary"}[5m]))))
+  expression: histogram_quantile(0.99, sum(rate(grpc_server_handling_seconds_bucket{grpc_service=~"etcdserverpb.*",grpc_type="unary"}[5m])))
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: same here.

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 9]
```diff
@@ -2836,8 +2834,7 @@ metadata:
     metric: grpc-request-slow-quantile
     source: rancher-monitoring
 spec:
-  expression: sum(histogram_quantile(0.99, sum(rate(grpc_server_handling_seconds_bucket{grpc_type="unary"}[5m]))))
-    by (instance)
+  expression: histogram_quantile(0.99, sum(rate(grpc_server_handling_seconds_bucket{grpc_service=~"etcdserverpb.*",grpc_type="unary"}[5m])) by (instance,le))
   legendFormat: Request slow([[instance]])
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: same here.

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 10]
```diff
@@ -2446,8 +2446,7 @@ metadata:
     metric: grpc-client-receive-bytes-sum-rate
     source: rancher-monitoring
 spec:
-  expression: sum(rate(etcd_network_client_grpc_received_bytes_total[5m]))  *
-    8 / 1024
+  expression: sum(rate(etcd_network_client_grpc_received_bytes_total[5m]))
   legendFormat: Client traffic in
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: also need to update the graph unit

File: pkg/controllers/user/monitoring/metricExpression.go (hunk 2 above)
Comment (thxCode) [reply pkg/controllers/user/monitoring/metricExpression.go]: @logan, Will the UI adapt to the change of `spec.yAxis.unit` as below?

```
apiVersion: management.cattle.io/v3
kind: ClusterMonitorGraph/ProjectMonitorGraph
metadata:
  ...
spec:
  ...
  yAxis:
    unit: bps
---
```

File: pkg/controllers/user/monitoring/metricExpression.go (hunk 10 above)
Comment (thxCode) [reply pkg/controllers/user/monitoring/metricExpression.go]: Which unit wants to update? The `graph: rpc-client-traffic` are all using `bps` now.

File: pkg/controllers/user/monitoring/metricExpression.go (hunk 1 above)
Comment (thxCode) [reply pkg/controllers/user/monitoring/metricExpression.go]: Rollbacked

File: pkg/controllers/user/monitoring/metricExpression.go (hunk 3 above)
Comment (thxCode) [reply pkg/controllers/user/monitoring/metricExpression.go]: Rollbacked

File: pkg/controllers/user/monitoring/metricExpression.go (hunk 4 above)
Comment (thxCode) [reply pkg/controllers/user/monitoring/metricExpression.go]: Yes, `sum(histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m])))` is a bad expression, `histogram_quantile(0.99, sum(rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m])))` is a wrong expression, I have committed the new one: `sum(histogram_quantile(0.99, sum(rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m])) by (instance, le)))`

File: pkg/controllers/user/monitoring/metricExpression.go (hunk 5 above)
Comment (thxCode) [reply pkg/controllers/user/monitoring/metricExpression.go]: This is a good expression. For one instance, `histogram_quantile` should be the outside function.

**Comment by aiwantaozi** [comment]
@thxCode since resource container doesn't have summary and details, could you update the metric, no need group by container name, and the legendFormat don't need [[container_name]] any more.

**Comment by thxCode** [comment]
@aiwantaozi, Updated

**Comment by aiwantaozi** [comment]
LGTM

**Comment by loganhz** [comment]
LGTM

**Comment by orangedeng** [comment]
LGTM

**Comment by thxCode** [comment]
Rebased & Squashed

**Comment by cjellick** [comment]
LGTM, restarting build bc it had an unrealted failure

**Pull request merged**